Fix `initialOptions` label rendering for integer based initial states

gehrisandro opened the pull request:
## Description

When the current state of a multi select contains integer based values which are not within the loaded options, instead of the label, the ID is rendered.

The reason for the issue is, that the value for all options is cast to a string by default: https://github.com/filamentphp/filament/blob/4.x/packages/forms/src/Components/Select.php#L152

If the state is now an array of integers, the value can not be found in the list of initial options:
https://github.com/filamentphp/filament/blob/4.x/packages/support/resources/js/utilities/select.js#L621

## Visual changes

Before:
<img width="348" height="366" alt="CleanShot 2025-08-24 at 07 38 51@2x" src="https://github.com/user-attachments/assets/6d113a28-3ee9-4977-a15e-c66d0cef7af8" />

After:
<img width="340" height="422" alt="CleanShot 2025-08-24 at 07 40 08@2x" src="https://github.com/user-attachments/assets/dd7d2671-049e-4e3a-9631-d64fd5f765ef" />


## Functional changes

- [x] Code style has been fixed by running the `composer cs` command.
- [x] Changes have been tested to not break existing functionality.
- [ ] Documentation is up-to-date. _(no change required)_

gehrisandro committed:
Ensure `missingValues` stores string representations of values

danharrin reviewed (changes requested):
I think we need to fix the value we pass to `initialOptions` instead. If the `$state` var is an array, we should map it to strings, if it is a singular value and not `blank()` we should also cast to a string. Then we don't need to accommodate in multiple places in the JS. Any thoughts? and does this fix the issue?

gehrisandro commented:
@danharrin Thank you for pointing out the handling in the `relationship()`. This actually helped me, to find the root cause of my problem.

What I did in Filament v3 is to fill the form with a list of IDs:
```php
$this->getSchema('form')->fill([
    'tags' => $this->taggable->userTags->pluck('id')->toArray(),
]);
```

In Filament v4 we now have to cast the IDs to strings:
```php
$this->getSchema('form')->fill([
    'tags' => $this->taggable->userTags->pluck('id')->map(fn (int $id): string => strval($id))->toArray(),
]);
```

I still think we should improve the JS to work with strings and integer based values, instead of casting the state to strings behind the scenes, which feels magically and unexpected to me.

If you think this isn't a relevant issue at all, feel free to close this PR.

danharrin commented:
I think we can make it easier by casting the state values to strings on the PHP side before they get passed to JS, probably somewhere in the view file. What do you think?

gehrisandro commented:
I gave it a try to cast the state to strings, but this isn't very easy. We would have to check for `Enums`, `Htmlable`, etc.
Basically everything that Laravel does here: https://github.com/laravel/framework/blob/12.x/src/Illuminate/Support/Js.php#L67

gehrisandro committed:
Ensure the state is a string or an array of strings when initializing the select

gehrisandro commented:
Updated the PR to convert the state in the select constructor.

gehrisandro committed:
Revert "Ensure the state is a string or an array of strings when initializing the select"

This reverts commit d41a546de2307416e63a3e66753b50068cb362de.

gehrisandro commented:
> I'm really sorry, I think I have changed my mind about the fix. If we change the `state` variable, it will queue a Livewire update when the field is loaded to convert the ints to strings, which isn't ideal if the field is live as it might make an immediate network request.
> 
> Based on your initial fix, are there any other places that need updating to work correctly with int/float keys instead of strings?

No problem. I've just reverted my commit from before to return to my initial solution.

As far as I am aware of, there are no other changes necessary.

danharrin commented:
Thanks, sorry for the extra work here
